lms/static/js/edxnotes/views/tabs/tags.js

```diff
@@ -0,0 +1,129 @@
+            title: 'Tags',
+            noTags: gettext('[no tags]'),  // User-defined tags cannot have spaces, so no risk of a collision.
+
+            renderContent: function () {
+                var notesByTag = {}, noTags = this.noTags, addNoteForTag, noteList, tags, i,
+                    sortedTagNames, container, group, noteGroup, tagTitle, titleMap;
+
+                // Iterate through all the notes and build up a dictionary structure by tag.
+                // Note that the collection will be in most-recently updated order already.
+                addNoteForTag = function (note, tag) {
+                    noteList = notesByTag[tag.toLowerCase()];
+                    if (noteList === undefined) {
+                        noteList = [];
+                        notesByTag[tag.toLowerCase()] = noteList;
+                    }
+                    // If a note was tagged with the same tag more than once, don't add again.
+                    // We can assume it would be the last element of the list because we iterate through
+                    // all tags on a given note before moving on to the text note.
+                    if (noteList.length === 0 || noteList[noteList.length - 1] !== note) {
+                        noteList.push(note);
+                    }
+                };
+
+                this.collection.each(function(note){
+                    tags = note.get('tags');
+                    if (tags.length === 0) {
+                        addNoteForTag(note, noTags);
+                    }
+                    else {
+                        for (i = 0; i < tags.length; i++) {
+                            addNoteForTag(note, tags[i]);
+                        }
+                    }
+                });
+
+                sortedTagNames = Object.keys(notesByTag).sort(function (a, b) {
+                    // "no tags" should always appear last
+                    if (a === noTags) {
+                        return 1;
+                    }
+                    else if (b === noTags) {
+                        return -1;
+                    }
+                    else if (notesByTag[a].length > notesByTag[b].length) {
+                        return -1;
+                    }
+                    else if (notesByTag[a].length < notesByTag[b].length) {
+                        return 1;
+                    }
+                    else {
+                        return a.toLowerCase() <= b.toLowerCase() ? -1 : 1;
+                    }
+                });
+
+               container = document.createDocumentFragment();
+
+                // Store map of titles for scrollToTag functionality.
+                this.titleMap = {};
+                titleMap = this.titleMap;
+
+                _.each(sortedTagNames, function (tagName) {
+                    noteGroup = notesByTag[tagName];
+                    var tagTitle = interpolate_text(
+                        gettext("{tagName} ({numberOfNotesWithTag})"),
```

Comment:
@cahrens I replaced the "2" with "٢" (which is how 2 is written in arabic) and therefore in the screenshot the order of everything is correct. If the count text was "2" it would look like "(social (2" if the user was using a RTL language. You noted this in a comment above too.